packages/components/data-resource/data-resource-create-form.tsx

```diff
@@ -94,14 +95,32 @@ const isFormValid = (form: CreateFormDataState) => {
     pvc,
     pvcFolderName,
   } = form;
-  return (
-    !!name?.trim() &&
-    name.length <= 42 &&
-    (((region || endpoint) &&
-      (secretName || (secretKey && accessKey)) &&
-      target) ||
-      (pvc && pvcFolderName))
-  );
+  const secretValid = secretName || (secretKey && accessKey);
+  let providerValid;
+  switch (provider) {
+    case BC_PROVIDERS.AWS: {
+      providerValid = region && secretValid && target;
+      break;
+    }
+    case BC_PROVIDERS.IBM:
+    case BC_PROVIDERS.S3: {
+      providerValid = endpoint && secretValid && target;
+      break;
+    }
+    case BC_PROVIDERS.AZURE: {
+      providerValid = secretValid && target;
+      break;
+    }
+    case BC_PROVIDERS.FILESYSTEM: {
+      providerValid = pvc && pvcFolderName;
+      break;
+    }
+    default: {
+      providerValid = '';
+      break;
+    }
+  }
+  return !!name?.trim() && name.length <= 42 && providerValid;
```

Comment:
return from switch case itself... no need to maintain `providerValid`

Reply:
also test this once thoroughly for all providers... make sure nothing breaks...

Reply:
I agree, return from the switch and have a variable at the top for a `nameValid`?

Reply:
yup